authorization/invitation/service/invitation_service_blackbox_test.go

```diff
@@ -34,619 +38,693 @@ func (s *invitationServiceBlackBoxTest) SetupTest() {
+		require.Error(s.T(), err)
+	})
 
-func (s *invitationServiceBlackBoxTest) TestIssueInvitationFailsForNonOwner() {
-	// Create a test user - this will be the owner
-	identity, err := test.CreateTestIdentityAndUserWithDefaultProviderType(s.DB, "invitationServiceBlackBoxTest-TestIssuingUser")
-	require.Nil(s.T(), err, "Could not create identity")
+	s.T().Run("should issue invitation for resource", func(t *testing.T) {
 
-	// Create an organization
-	orgId, err := s.orgService.CreateOrganization(s.Ctx, identity.ID, "Test Organization "+uuid.NewV4().String())
-	require.Nil(s.T(), err, "Could not create organization")
+		g := s.NewTestGraph()
 
-	// Create another test user - we will attempt to have this user invite themselves to the organization
-	otherIdentity, err := test.CreateTestIdentityAndUserWithDefaultProviderType(s.DB, "invitationServiceBlackBoxTest-TestInviteeUser")
-	require.Nil(s.T(), err, "Could not create other identity")
+		// Create a test user - this will be the inviter
+		inviter := g.CreateUser()
 
-	invitations := []invitation.Invitation{
-		{
-			IdentityID: &otherIdentity.ID,
-			Member:     true,
-		},
-	}
+		// Create another test user - we will invite this one to accept a role for the resource
+		invitee := g.CreateUser()
+		inviteeID := invitee.IdentityID()
 
-	err = s.Application.InvitationService().Issue(s.Ctx, otherIdentity.ID, orgId.String(), invitations)
-	require.Error(s.T(), err)
-}
+		space := g.CreateSpace()
+		space.AddAdmin(inviter)
 
-func (s *invitationServiceBlackBoxTest) TestIssueInvitationFailsForUnknownUser() {
-	// Create a test user - this will be the organization owner
-	identity, err := test.CreateTestIdentityAndUserWithDefaultProviderType(s.DB, "invitationServiceBlackBoxTest-TestIssuingUser")
-	require.Nil(s.T(), err, "Could not create identity")
+		// Create an invitation
+		invitations := []invitation.Invitation{
+			{
+				IdentityID: &inviteeID,
+				Roles:      []string{"admin"},
+			},
+		}
 
-	// Create an organization
-	orgId, err := s.orgService.CreateOrganization(s.Ctx, identity.ID, "Test Organization "+uuid.NewV4().String())
-	require.Nil(s.T(), err, "Could not create organization")
+		var messages []notification.Message
+		*s.notificationServiceMock = *testservice.NewNotificationServiceMock(s.T())
+		s.notificationServiceMock.SendMessagesAsyncFunc = func(p context.Context, p1 []notification.Message) (r2 error) {
+			messages = p1
+			return nil
+		}
 
-	invalidIdentityID := uuid.NewV4()
+		// Issue the invitation
+		err := s.Application.InvitationService().Issue(s.Ctx, inviter.IdentityID(), space.SpaceID(), invitations)
+		require.NoError(s.T(), err)
 
-	invitations := []invitation.Invitation{
-		{
-			IdentityID: &invalidIdentityID,
-			Member:     true,
-		},
-	}
+		require.Equal(s.T(), uint64(1), s.notificationServiceMock.SendMessagesAsyncCounter)
+		require.Equal(s.T(), inviteeID.String(), messages[0].TargetID)
```

Comment:
you could also add a `require.Len(t, messages, 1)` to make sure that a single message was passed to the mock NotificationService, before calling `messages[0]`